config/v1/types_cluster_version.go

```diff
@@ -215,6 +222,82 @@ type UpdateHistory struct {
+// +kubebuilder:validation:Enum=openshift-samples
+type ClusterVersionCapability string
+
+const (
+	// ClusterVersionCapabilityOpenShiftSamples manages the sample
+	// image streams and templates stored in the openshift
+	// namespace, and any registry credentials, stored as a secret,
+	// needed for the image streams to import the images they
+	// reference.
+	ClusterVersionCapabilityOpenShiftSamples ClusterVersionCapability = "openshift-samples"
+)
+
+// ClusterVersionCapabilitySet defines sets of cluster version capabilities.
+// +kubebuilder:validation:Enum=None;v4.11;vCurrent
+type ClusterVersionCapabilitySet string
+
+const (
+	// ClusterVersionCapabilitySetNone is an empty set containing no capabilities.
+	ClusterVersionCapabilitySetNone ClusterVersionCapabilitySet = "None"
+
+	// ClusterVersionCapabilitySet4_11 is the recommended set of
+	// optional capabilities for the 4.11 version of OpenShift.
+	// This list will remain the same no matter which version of
+	// OpenShift is installed.
+	ClusterVersionCapabilitySet4_11 ClusterVersionCapabilitySet = "v4.11"
+
+	// ClusterVersionCapabilitySetCurrent is the recommended set of
+	// optional capabilities for the current version of OpenShift.
+	ClusterVersionCapabilitySetCurrent ClusterVersionCapabilitySet = "vCurrent"
+)
+
+// ClusterVersionCapabilitySets defines sets of cluster version capabilities.
+var ClusterVersionCapabilitySets = map[ClusterVersionCapabilitySet][]ClusterVersionCapability{
+	ClusterVersionCapabilitySetNone: {},
+	ClusterVersionCapabilitySet4_11: {
+		ClusterVersionCapabilityOpenShiftSamples,
+	},
+	ClusterVersionCapabilitySetCurrent: {
+		ClusterVersionCapabilityOpenShiftSamples,
+	},
+}
+
+// ClusterVersionCapabilitiesSpec selects the managed set of
+// optional, core cluster components.
+// +k8s:deepcopy-gen=true
+type ClusterVersionCapabilitiesSpec struct {
+	// baselineCapabilitySet selects an initial set of
+	// capabilities, which can be extended via
+	// additionalEnabledCapabilities.
+	// +kubebuilder:default:="vCurrent"
+	BaselineCapabilitySet ClusterVersionCapabilitySet `json:"baselineCapabilitySet,omitempty"`
+
+	// additionalEnabledCapabilities extends the set of managed
+	// capabilities beyond the baseline defined in
+	// baselineCapabilitySet.
+	// +listType=atomic
+	// +optional
+	AdditionalEnabledCapabilities []ClusterVersionCapability `json:"additionalEnabledCapabilities,omitempty"`
+}
+
+// ClusterVersionCapabilitiesStatus describes the state of optional,
+// core cluster components.
+// +k8s:deepcopy-gen=true
+type ClusterVersionCapabilitiesStatus struct {
+	// enabledCapabilities lists all the capabilities that are currently managed.
+	// +listType=atomic
+	// +optional
+	EnabledCapabilities []string `json:"enabledCapabilities,omitempty"`
+
+	// knownCapabilities lists all the capabilities known to the current cluster.
+	// +listType=atomic
+	// +optional
+	KnownCapabilities []string `json:"knownCapabilities,omitempty"`
```

Comment:
if we later add default on and default off, how will we represent it using this API?

Reply:
you mean default on and default off for individual capabilities?

Reply:
folks with a cluster running don't need to care about whether caps were default on or off.  The status struct will tell them the list of all enumerated cap strings here, the set that are enabled in their cluster in `enabledCapabilities`, and (via `knownCapabilities` - `enabledCapabilities` set subtraction), the strings they could inject into `additionalEnabledCapabilities` if they want more stuff.  They don't have to guess about options for altering `baselineCapabilitySet`, although I guess we could dump that API map into ClusterVersion's status too, if folks wanted that as a later extension.  But personally, I expect folks who wonder about the specifics of the various capability sets to be opinionated enough to want to use `additionalEnabledCapabilities`.

Reply:
> you mean default on and default off for individual capabilities?

yes.

> although I guess we could dump that API map into ClusterVersion's status too, if folks wanted that as a later extension

how would we do this?

Reply:
I think I can see how to do this by adding a capabilitySet indexed slice that lists what it enables.  That sort of consideration is what I was hoping for.